test/rules/no-tautology-expression/test.ts.lint

```diff
@@ -0,0 +1,108 @@
+const expr = 123 === 123;
+             ~~~~~~~~~~~  [Expression is either a tautology or a contradiction. Binary expression is redundant.]
+const someVar = 100;
+const expr = someVar === someVar;
+             ~~~~~~~~~~~~~~~~~~~  [Expression is either a tautology or a contradiction. Binary expression is redundant.]
+
+const someFunc = () => {
+    if ("SomeStr" === "SomeStr") {
+        ~~~~~~~~~~~~~~~~~~~~~~~  [Expression is either a tautology or a contradiction. Binary expression is redundant.]
+        return true;
+    }
+}
+
+const someFunc = () => {
+    if (100 === 100) {
+        ~~~~~~~~~~~  [Expression is either a tautology or a contradiction. Binary expression is redundant.]
+        return true;
+    }
+}
+
+const someFunc = () => {
+    let someVar = 100;
+    if (someVar === someVar) {
+        ~~~~~~~~~~~~~~~~~~~  [Expression is either a tautology or a contradiction. Binary expression is redundant.]
+        return true;
+    }
+}
+
+const someFunc = () => {
+    if (1 !== 1) {
+        ~~~~~~~  [Expression is either a tautology or a contradiction. Binary expression is redundant.]
+        return true;
+    }
+}
+
+const someFunc = () => {
+    if (1 > 1) {
+        ~~~~~  [Expression is either a tautology or a contradiction. Binary expression is redundant.]
+        return true;
+    }
+}
+
+const someFunc = () => {
+    const someVar = 123;
+    if (someVar < someVar) {
+        ~~~~~~~~~~~~~~~~~  [Expression is either a tautology or a contradiction. Binary expression is redundant.]
+        return true;
+    }
+}
+
+const someFunc = () => {
+    if ("str" == "str") {
+        ~~~~~~~~~~~~~~ [Expression is either a tautology or a contradiction. Binary expression is redundant.]
+        return true;
+    }
+}
+
+const someFunc = () => {
+    if (123 != 123) {
+        ~~~~~~~~~~  [Expression is either a tautology or a contradiction. Binary expression is redundant.]
+        return true;
+    }
+}
+
+const someFunc = () => {
+    if ("str" <= "str") {
+        ~~~~~~~~~~~~~~  [Expression is either a tautology or a contradiction. Binary expression is redundant.]
+        return true;
+    }
+}
+
+const someFunc = () => {
+    if ("str" >= "str") {
+        ~~~~~~~~~~~~~~  [Expression is either a tautology or a contradiction. Binary expression is redundant.]
+        return true;
+    }
+}
+
+const someFunc = () => {
+    const someVar = true;
+    if (someVar || someVar) {
+        ~~~~~~~~~~~~~~~~~~  [Expression is either a tautology or a contradiction. Binary expression is redundant.]
+        return true;
+    }
+}
+
+const someFunc = () => {
+    const someVar = true;
+    if (someVar && someVar) {
+        ~~~~~~~~~~~~~~~~~~  [Expression is either a tautology or a contradiction. Binary expression is redundant.]
+        return true;
+    }
+}
+
+const someVar1 = 3 + 3;
+const someVar2 = 3 - 3;
+const someVar3 = 3 * 3;
+const someVar4 = 3 % 3;
+const someVar5 = 3 / 3;
```

Comment:
Were you planning on using these or the following blank lines for something? If not, let's remove them as unused.

Reply:
These are meant for testing that arithmetic binary expressions are not being checked by the rule.
Still wanna remove them?